G1ANT.Addon.Selenium/SeleniumAddon.cs

```diff
@@ -28,79 +25,32 @@ public class SeleniumAddon : Language.Addon
-            UnpackMSEdgeDriver();
             base.LoadDlls();
         }
 
         private string UnpackFolder { get => AbstractSettingsContainer.Instance.UserDocsAddonFolder.FullName; }
         
-        private void UnpackMSEdgeDriver()
-        {
-            var filename = "msedgedriver.exe";
-            var driverData = Environment.Is64BitOperatingSystem ? Resources.msedgedriver_64 : Resources.msedgedriver_32;
-
-            if (DriverNeedsUpdate(filename, driverData))
-                UpdateDriverFromResource(filename, driverData);
-        }
-
         private void UnpackDrivers()
         {
-            var embeddedResourceDictionary = new Dictionary<string, byte[]>()
-            {
-                { "chromedriver.exe", Resources.chromedriver },
-                { "geckodriver.exe", Resources.geckodriver },
-                { "IEDriverServer.exe", Resources.IEDriverServer }
+            var driverResources = new List<SeleniumDriverResourceDescription>()
+            { 
+                new SeleniumDriverResourceDescription("chromedriver.exe", "chromedriver"),
+                new SeleniumDriverResourceDescription("geckodriver.exe", "geckodriver_32", "geckodriver_64"),
+                new SeleniumDriverResourceDescription("IEDriverServer.exe", "IEDriverServer"),
+                new SeleniumDriverResourceDescription("msedgedriver.exe", "msedgedriver_32", "msedgedriver_64")
             };
-            foreach (var embededResource in embeddedResourceDictionary.Where(e => DriverNeedsUpdate(e.Key, e.Value)))
-            {
-                UpdateDriverFromResource(embededResource.Key, embededResource.Value);
-            }
-        }
-
-        private bool DriverNeedsUpdate(string filename, byte[] resourceData)
-        {
-            return !DoesFileExist(UnpackFolder, filename) || !AreFilesOfTheSameLength(resourceData.Length, UnpackFolder, filename);
-        }
-
-        private void UpdateDriverFromResource(string filename, byte[] resourceData)
-        {
-            try
-            {
-                KillWorkingProcess(Path.GetFileNameWithoutExtension(filename));
-                using (FileStream stream = File.Create(Path.Combine(UnpackFolder, filename)))
-                {
-                    stream.Write(resourceData, 0, resourceData.Length);
-                }
-            }
-            catch (Exception ex)
-            {
-                RobotMessageBox.Show(ex.Message);
-            }
-        }
 
-        private void KillWorkingProcess(string processName)
-        {
-            foreach (Process proc in Process.GetProcessesByName(processName))
+            foreach (var driver in driverResources)
             {
                 try
                 {
-                    proc.Kill();
+                    driver.UnpackIfNeeded(Assembly, UnpackFolder);
                 }
-                catch
+                catch (Exception ex)
                 {
-
+                    RobotMessageBox.Show(ex.Message);
```

Comment:
Maybe we should write that information to the log?

Reply:
This implementation has been moved from SeleniumAddon, I don't want to change the code that works fine.

Reply:
I don't want to change the behaviour that was already implemented.